src/dune_rules/warning.ml

```diff
@@ -0,0 +1,78 @@
+open Import
+
+module Name = struct
+  include String
+
+  include Stringlike.Make (struct
+      type t = string
+
+      let module_ = "Warning"
+      let description = "Warning name"
+      let description_of_valid_string = None
+      let hint_valid = None
+      let of_string_opt x = if Dune_sexp.Atom.is_valid x then Some x else None
+      let to_string x = x
+    end)
+end
+
+type t =
+  { name : Name.t
+  ; since : Syntax.Version.t
+  ; default : Syntax.Version.t -> Config.Toggle.t
+  }
+
+let frozen = ref false
+let all : (Name.t, t) Table.t = Table.create (module Name) 12
```

Comment:
12?

Reply:
Any number would do. We just need some initial table size.